src/v0/destinations/tiktok_ads/data/TikTokTrackV2.json

```diff
@@ -0,0 +1,128 @@
+[
+  {
+    "destKey": "event_id",
+    "sourceKeys": ["properties.eventId", "properties.event_id", "messageId"],
+    "required": false
+  },
+  {
+    "destKey": "event_time",
+    "sourceKeys": "timestamp",
+    "sourceFromGenericMap": true,
+    "required": true,
+    "metadata": {
+      "type": "secondTimestamp"
+    }
+  },
+  {
+    "destKey": "test_event_code",
+    "sourceKeys": "properties.testEventCode",
+    "required": false
+  },
+  {
+    "destKey": "limited_data_use",
+    "sourceKeys": "properties.limited_data_use",
+    "required": false
+  },
+  {
+    "destKey": "properties.contents",
+    "sourceKeys": "properties.contents",
+    "required": false
+  },
+  {
+    "destKey": "properties.content_type",
+    "sourceKeys": ["properties.contentType", "properties.content_type"],
+    "metadata": {
+      "defaultValue": "product"
+    }
+  },
+  {
+    "destKey": "properties.shop_id",
+    "sourceKeys": ["properties.shop_id", "properties.shopId"],
+    "required": false,
+    "metadata": {
+      "type": "toString"
+    }
+  },
+  {
+    "destKey": "properties.order_id",
+    "sourceKeys": ["properties.order_id", "properties.orderId"],
+    "required": false,
+    "metadata": {
+      "type": "toString"
+    }
+  },
+  {
+    "destKey": "properties.currency",
+    "sourceKeys": "properties.currency",
+    "required": false
+  },
+  {
+    "destKey": "properties.value",
+    "sourceKeys": "properties.value",
+    "required": false
+  },
+  {
+    "destKey": "properties.description",
+    "sourceKeys": "properties.description",
+    "required": false
+  },
+  {
+    "destKey": "properties.query",
+    "sourceKeys": "properties.query",
+    "required": false
+  },
+  {
+    "destKey": "page.url",
+    "sourceKeys": ["properties.url", "context.page.url"],
+    "required": true
+  },
+  {
+    "destKey": "page.referrer",
+    "sourceKeys": ["properties.referrer", "context.page.referrer"],
+    "required": false
+  },
+  {
+    "destKey": "user.locale",
+    "sourceKeys": "context.locale",
+    "required": false
+  },
+  {
+    "destKey": "user.ttclid",
+    "sourceKeys": "properties.ttclid",
+    "required": false
+  },
+  {
+    "destKey": "user.ttp",
+    "sourceKeys": "properties.ttp",
+    "required": false
+  },
+  {
+    "destKey": "user.email",
+    "sourceKeys": [
```

Comment:
Can't we take it from generic mapping?

Reply:
Generic mapping for email take some value from externalId as well thats why did not took it from there

Reply:
Also it wont have `context.user.email` property there

Reply:
That externalId is specifically for rETL sources. But yes from other sources also we can get email there which is very unlikely as it is not documented anywhere. We generally don't get email inside `context.user`. Why we are including that now?

Reply:
We support `context.user` in case of tiktok_Ads somehow from previous version. Shall we remove it ?

Reply:
Got it. Let it be this way then.